src/task.c

```
@@ -8,32 +8,55 @@
 #define CONTEXT_SIZE 8
 
 static uint32_t *tasks[TASK_LIMIT];
+static uint32_t *stacks[TASK_LIMIT];
+static int      cur_task;
+
+void task_delete()
+{
+    uint32_t *temp = tasks[cur_task];
+
+    for (int i = cur_task; tasks[i] && i < TASK_LIMIT - 1; i++)
+    {
+        tasks[i] = tasks[i + 1];
+        stacks[i] = stacks[i + 1];
+    }
+    tasks[TASK_LIMIT - 1] = NULL;
+    stacks[TASK_LIMIT - 1] = NULL;
+    free(temp);
+    exit();
+}
 
 void    task_add(void (*task_code)(void), size_t stack_size)
 {
     int i = 0;
 
-    while (tasks[i])
+    while (tasks[i] && i < TASK_LIMIT)
         i++;
+
+    if (i == TASK_LIMIT)
+        return ;
```

maybe return an error code?